main.go

```diff
@@ -46,12 +50,29 @@ func WithError(next func(w http.ResponseWriter, r *http.Request) (int, error)) f
 	return fn
 }
 
+func GenerateNonce() (string, error) {
+	// Generate a random nonce to include in our challenge
+	nonceBytes := make([]byte, 32)
+	n, err := rand.Read(nonceBytes)
+	if n != 32 {
+		return "", errors.New("nonce: n != 64 (bytes)")
+	} else if err != nil {
+		return "", err
+	}
+	nonce := hex.EncodeToString(nonceBytes)
+	return nonce, nil
+}
+
 func RequestNonceHandler() func(w http.ResponseWriter, r *http.Request) (int, error) {
 	fn := func(w http.ResponseWriter, r *http.Request) (int, error) {
-		return http.StatusNotImplemented, terror.ErrNotImplemented
+		nonce, err := GenerateNonce()
+		println(nonce, err)
+		w.Write([]byte(nonce))
```

Comment:
I'd like a more userfriendly message to be signed as a Typed Message:

![image](https://user-images.githubusercontent.com/4482567/120435830-c6afc300-c3b0-11eb-8314-8f4313c4716c.png)

```
// Example payload fields
type Payload struct {
    NonceHex string `json:"nonceHex"`
    Message string `json:"message"`
    Address string `json:"address"`
}

func GenerateNonce(message) (*Payload, error) {
    ...
}
```

This way we can make the UX a little better and readable to not scare away users.